### Data Scientist Job Market in the U.S

To help job seekers to better understand the job market, we are going to analyse the 'Data Scientist Job Market in the U.S.' dataset. You can find the data <a href="https://www.kaggle.com/sl6149/data-scientist-job-market-in-the-us">here. </a>The information ic collected from Indeed and includes: Company Name, Position Name, Location, Job Description, and Number of Reviews of the Company.

In [ ]:
import pandas as pd 
import numpy as np
from matplotlib import pyplot as plt
import seaborn

In [225]:
#import the dataset that contains the data for job market round the US
df = pd.read_csv('alldata.csv')

In [226]:
df.head()

position             company  \
0                               Development Director             ALS TDI   
1  An Ostentatiously-Excitable Principal Research...  The Hexagon Lavish   
2                                     Data Scientist      Xpert Staffing   
3                                       Data Analyst      Operation HOPE   
4  Assistant Professor -TT - Signal Processing & ...    Emory University   

                                         description  reviews  \
0  Development Director\nALS Therapy Development ...      NaN   
1  Job Description\n\n"The road that leads to acc...      NaN   
2  Growing company located in the Atlanta, GA are...      NaN   
3  DEPARTMENT: Program OperationsPOSITION LOCATIO...     44.0   
4  DESCRIPTION\nThe Emory University Department o...    550.0   

             location  
0  Atlanta, GA 30301   
1         Atlanta, GA  
2         Atlanta, GA  
3  Atlanta, GA 30303   
4         Atlanta, GA

In [227]:
#import the datasets that contains the data for the job market in specific regions in case we'll need it
dfAL = pd.read_csv('fulltimeAL.csv')
dfAT = pd.read_csv('fulltimeAL.csv')
dfBO = pd.read_csv('fulltimeAL.csv')
dfBOS = pd.read_csv('fulltimeAL.csv')
dfCHI = pd.read_csv('fulltimeAL.csv')
dfDC = pd.read_csv('fulltimeAL.csv')
dfLA = pd.read_csv('fulltimeAL.csv')
dfMA = pd.read_csv('fulltimeAL.csv')
dfMV = pd.read_csv('fulltimeAL.csv')
dfNY = pd.read_csv('fulltimeAL.csv')
dfRM = pd.read_csv('fulltimeAL.csv')
dfSD = pd.read_csv('fulltimeAL.csv')
dfSEA = pd.read_csv('fulltimeAL.csv')
dfSF = pd.read_csv('fulltimeAL.csv')
dfSU = pd.read_csv('fulltimeAL.csv')

In [228]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6964 entries, 0 to 6963
Data columns (total 5 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   position     6953 non-null   object 
 1   company      6953 non-null   object 
 2   description  6953 non-null   object 
 3   reviews      5326 non-null   float64
 4   location     6953 non-null   object 
dtypes: float64(1), object(4)
memory usage: 272.2+ KB


Let's find the number of missing values

In [229]:
df.isnull().sum()

position         11
company          11
description      11
reviews        1638
location         11
dtype: int64

Reviews have 1638 null values, so nearly 24% of its data is missing.



In [230]:
df.head()

position             company  \
0                               Development Director             ALS TDI   
1  An Ostentatiously-Excitable Principal Research...  The Hexagon Lavish   
2                                     Data Scientist      Xpert Staffing   
3                                       Data Analyst      Operation HOPE   
4  Assistant Professor -TT - Signal Processing & ...    Emory University   

                                         description  reviews  \
0  Development Director\nALS Therapy Development ...      NaN   
1  Job Description\n\n"The road that leads to acc...      NaN   
2  Growing company located in the Atlanta, GA are...      NaN   
3  DEPARTMENT: Program OperationsPOSITION LOCATIO...     44.0   
4  DESCRIPTION\nThe Emory University Department o...    550.0   

             location  
0  Atlanta, GA 30301   
1         Atlanta, GA  
2         Atlanta, GA  
3  Atlanta, GA 30303   
4         Atlanta, GA

Separate the city and state:

In [231]:
separ = df.location.str.split(',', expand = True)
df['city'] = separ[0]
df['newstate'] = separ[1]

print(df.city)

state1 = df.newstate.str.split(expand = True)
df['state'] = state1[0]
print(df.state)



0         Atlanta
1         Atlanta
2         Atlanta
3         Atlanta
4         Atlanta
          ...    
6959    Sunnyvale
6960    Sunnyvale
6961    Sunnyvale
6962    Sunnyvale
6963    Sunnyvale
Name: city, Length: 6964, dtype: object
0       GA
1       GA
2       GA
3       GA
4       GA
        ..
6959    CA
6960    CA
6961    CA
6962    CA
6963    CA
Name: state, Length: 6964, dtype: object


In [232]:
#drop the column location:
df = df.drop(['location'], axis=1)
df = df.drop(['newstate'], axis=1)

In [233]:
df.head(5)

position             company  \
0                               Development Director             ALS TDI   
1  An Ostentatiously-Excitable Principal Research...  The Hexagon Lavish   
2                                     Data Scientist      Xpert Staffing   
3                                       Data Analyst      Operation HOPE   
4  Assistant Professor -TT - Signal Processing & ...    Emory University   

                                         description  reviews     city state  
0  Development Director\nALS Therapy Development ...      NaN  Atlanta    GA  
1  Job Description\n\n"The road that leads to acc...      NaN  Atlanta    GA  
2  Growing company located in the Atlanta, GA are...      NaN  Atlanta    GA  
3  DEPARTMENT: Program OperationsPOSITION LOCATIO...     44.0  Atlanta    GA  
4  DESCRIPTION\nThe Emory University Department o...    550.0  Atlanta    GA

In [263]:
#20 Companies that have the most reviews
top_review = df.groupby(['company'])['reviews'].sum().reset_index()
top_review.sort_values(['reviews'], ascending = False).head(10)



company    reviews
122        Amazon.com  9246299.0
2100          Walmart   740478.0
1270        Microsoft   632613.0
1066  JP Morgan Chase   548616.0
1925   The Home Depot   465284.0
232   Bank of America   395085.0
987               IBM   356028.0
866            Google   331488.0
57          Accenture   312390.0
1858           TARGET   299475.0

In [264]:
#20 companies that have the most open positions

positions_company = df.groupby(['company'])['position'].count().sort_values(ascending = False).reset_index()
positions_company.head(10)

company  position
0                              Amazon.com       358
1                          Ball Aerospace       187
2                               Microsoft       137
3                                  Google       134
4                      NYU Langone Health        77
5  Fred Hutchinson Cancer Research Center        70
6                                    KPMG        66
7                                  Lab126        50
8                                Facebook        49
9                         Broad Institute        49

In [236]:

most_positions = pd.DataFrame(df['position'].value_counts())
most_positions

#Another way using of counting them
#df.groupby(['position'])['company'].agg(['count']).sort_values(by='count', ascending=False).reset_index()

position
Data Scientist                                           351
Senior Data Scientist                                     96
Research Analyst                                          64
Data Engineer                                             60
Machine Learning Engineer                                 56
...                                                      ...
Applied Scientist- Zillow Offers Team                      1
Remote Sensing Data Scientist (PhD)                        1
Data Scientist: Statistics                                 1
Service Desk Specialist I - Field Science and R...         1
Sr. Applied Scientist, Sensors                             1

[5242 rows x 1 columns]

In [237]:
cities_most_jobs = pd.DataFrame(df['city'].value_counts())
cities_most_jobs.head(10)

city
New York        848
Seattle         777
Cambridge       694
Boston          629
San Francisco   564
Chicago         471
San Diego       412
Washington      340
Mountain View   277
Atlanta         269

In [238]:
states_most_jobs = pd.DataFrame(df['state'].value_counts())
states_most_jobs.head(10)

state
CA   2152
MA   1323
WA    935
NY    926
IL    471
DC    340
GA    269
CO    251
TX    218
NJ     68

In [239]:
#Now let's find how many types of positions are ther in the position variable:
position_nr = df['position'].value_counts()
position_nr.head(10)

Data Scientist               351
Senior Data Scientist         96
Research Analyst              64
Data Engineer                 60
Machine Learning Engineer     56
Lead Data Scientist           31
Research Scientist            30
Sr. Data Scientist            28
Principal Data Scientist      24
Software Engineer             22
Name: position, dtype: int64

In [255]:
df['data_analysts'] = df.position.apply(lambda x: 'Data Analyst' in str(x))

print(df['data_analysts'].value_counts())


False    6840
True      124
Name: data_analysts, dtype: int64


In [262]:
#check the new column
df.head()

position             company  \
0                               Development Director             ALS TDI   
1  An Ostentatiously-Excitable Principal Research...  The Hexagon Lavish   
2                                     Data Scientist      Xpert Staffing   
3                                       Data Analyst      Operation HOPE   
4  Assistant Professor -TT - Signal Processing & ...    Emory University   

                                         description  reviews     city state  \
0  Development Director\nALS Therapy Development ...      NaN  Atlanta    GA   
1  Job Description\n\n"The road that leads to acc...      NaN  Atlanta    GA   
2  Growing company located in the Atlanta, GA are...      NaN  Atlanta    GA   
3  DEPARTMENT: Program OperationsPOSITION LOCATIO...     44.0  Atlanta    GA   
4  DESCRIPTION\nThe Emory University Department o...    550.0  Atlanta    GA   

   data_analysts  
0          False  
1          False  
2          False  
3           True  
4          False

In [258]:
123/6840*100
#So only 1.8% of the job positions are Data Analysts

1.7982456140350878